routers/repo/branch.go

```diff
@@ -213,11 +221,68 @@ func loadBranches(ctx *context.Context) []*Branch {
 			ctx.ServerError("GetLatestPullRequestByHeadInfo", err)
 			return nil
 		}
+		mergeMovedOn := false
 		if pr != nil {
 			if err := pr.LoadIssue(); err != nil {
 				ctx.ServerError("pr.LoadIssue", err)
 				return nil
 			}
+			if repo, ok := repoIDToRepo[pr.BaseRepoID]; ok {
+				pr.HeadRepo = repo
+			} else if err := pr.LoadBaseRepo(); err != nil {
+				ctx.ServerError("pr.LoadBaseRepo", err)
+				return nil
+			} else {
+				repoIDToRepo[pr.BaseRepoID] = pr.BaseRepo
+			}
+
+			if repo, ok := repoIDToRepo[pr.HeadRepoID]; ok {
+				pr.HeadRepo = repo
+			} else if err := pr.LoadHeadRepo(); err != nil {
+				ctx.ServerError("pr.LoadHeadRepo", err)
+				return nil
+			} else {
+				repoIDToRepo[pr.HeadRepoID] = pr.HeadRepo
+			}
+
+			if pr.HasMerged && pr.HeadRepo != nil {
```

Comment:
Can `pr.HeadRepo` be `nil` here?

Reply:
Well... not the way I'd written the code... 

It can now.

Reply:
Line 219 only gets the PRs by HeadInfo - that is those that are in the head repo. Therefore the head repo is always going to be the ctx.Repo.Repository.